datafusion/functions-array/src/make_array.rs

```diff
@@ -131,6 +131,11 @@ impl ScalarUDFImpl for MakeArray {
     }
 }
 
+// Empty array is a special case that is useful for many other array functions
+pub(super) fn empty_array_type() -> DataType {
+    DataType::List(Arc::new(Field::new("item", DataType::Int64, true)))
+}
```

Comment:
I don't get the rationale behind that. From type-theoretical point of view, an empty array has type `Array<Null>` because you can coerce `Null` to any other type.

Reply:
You can think of this a type that is coerced from null to i64. You can't run the kernel function with null type, you need to coerce it to other type, and i64 is the default type we define

Reply:
> You can't run the kernel function with null type

That's the real issue. Coercing to i64 is just a band-aid that creates other problems.

Reply:
continue on https://github.com/apache/datafusion/pull/15149#issuecomment-2726151394, let's discussion about the issue it has